platforms/software/dependency-management/src/main/java/org/gradle/internal/component/resolution/failure/describer/ResolutionFailureDescriber.java

```diff
@@ -48,11 +48,12 @@ default boolean canDescribeFailure(FAILURE failure) {
      * for the given failure.
      *
      * @param failure the failure to describe
-     * @param schema the schema of the attributes that were used to select the variant, if available
+     * @param attributeDescribers describes individual attributes
+     *
      * @return the exception that describes the failure
      *
      * @implSpec Testing {@link #canDescribeFailure(ResolutionFailure)} should <strong>NOT</strong> be done by
      * implementations of this method; ensuring this is done first is the responsibility of the caller.
      */
-    AbstractResolutionFailureException describeFailure(FAILURE failure, @SuppressWarnings("OptionalUsedAsFieldOrParameterType") Optional<AttributesSchemaInternal> schema);
+    AbstractResolutionFailureException describeFailure(FAILURE failure, List<AttributeDescriber> attributeDescribers);
```

Comment:
Agreed. This was just following the prior pattern where the attributes schema was already passed into each describer -- even if they were not needed. 

For the sake of scope, I'd prefer to not expand the changes in this PR more than necessary. I think its _better_ now that we are by default passing less data to each describer than before. Before we passed the whole schema, now we pass a subset of the data that that schema provided.

Do you think it would be okay to push a greater refactoring to some future work?

Reply:
It's fine with me to split this into a separate change, as long as we do it relatively soon.  I agree this is probably already a slight improvement.  Can you make an issue for 8.11?